### PR TITLE
Sanity check for source and destination row counts

### PR DESCRIPTION
**Pull request by stdavis**
## Description of Changes
Adds a check between `changes.total_rows` (source) and the `crate.destination` row count after an update has been run. If they don't match it sets the crate status to `WARNING` with a helpful message. You can test it by adding features to the destination of a crate and then running forklift.

Here's what the console report looks like:
image

and the email:
image

This will be helpful in identifying and diagnosing the duplicate features in the destination that we have seen lately.

### Test results and coverage

```
Name                     Stmts   Miss     Cover   Missing
---------------------------------------------------------
forklift\arcgis.py          65      3    95.38%   48, 96-98
forklift\cli.py            224     64    71.43%   138, 173-178, 185, 232-235, 283-300, 307-318, 328-377

forklift\core.py           244      9    96.31%   84-88, 155, 196, 204, 363, 438
forklift\lift.py           131     26    80.15%   45-46, 134-155, 161, 179-180, 216-218, 231-232
forklift\models.py         169      4    97.63%   75, 318, 353-354
---------------------------------------------------------
TOTAL                      923    106    88.52%

5 files skipped due to complete coverage.
-----------------------------------------------------------------------------
151 tests run in 448.876 seconds (151 tests passed)

lint: commands succeeded
  congratulations :)
```

### Speed test results

```
Speed Test Results
Dry Run: 5.78 minutes
Repeat: 1.96 minutes
```

### CODE REVIEW

**Comment by steveoh** [comment]
Oh I like this

File: src/forklift/core.py [hunk 1]
```diff
@@ -483,3 +488,27 @@ def _is_naughty_field(fld):
...
+    changes: Changes
+
+    Validates that the row counts between source and destination are the same (ignoring empty geometries)
+
+    returns: (valid, message)
+        valid: Boolean - true if counts match
+        message: String - warning message if any
+    '''
+
+    destination_rows = len([row for row in arcpy.da.SearchCursor(crate.destination, ['*'])])
```
Comment (steveoh) [inline src/forklift/core.py]: Is this faster than `arcpy.GetCount_management()`?

Comment (stdavis) [inline src/forklift/core.py]: Yes, I believe so. But haven't tested...

Comment (stdavis) [inline src/forklift/core.py]: Oh. OK. I'll switch it.

Comment (stdavis) [inline src/forklift/core.py]: Switched to `arcpy.GetCount()`.

**Comment by stdavis** [comment]
I deployed this as v5.1.0 this morning.

**Pull request merged**